### PR TITLE
Basic getting started example

### PR DESCRIPTION
**Pull request by markpollack**
Note, had to make the following changes to the pom generated from the current initialzr site.

----
		
			
			
			
		

		<dependency>
			<groupId>org.springframework.cloud</groupId>
			<artifactId>spring-cloud-stream-codec</artifactId>
			<version>1.0.0.BUILD-SNAPSHOT</version>
		</dependency>

		<dependency>
			<groupId>org.springframework.cloud</groupId>
			<artifactId>spring-cloud-stream</artifactId>
			<version>1.0.0.BUILD-SNAPSHOT</version>
		</dependency>

		<dependency>
			<groupId>org.springframework.cloud</groupId>
			<artifactId>spring-cloud-stream-binder-kafka</artifactId>
			<version>1.0.0.BUILD-SNAPSHOT</version>
		</dependency>

----

### CODE REVIEW

**Comment by dsyer** [comment]
>  had to make the following changes to the pom generated from the current initialzr site

This is fubar, and it's not the initializr's fault. We need to change the starter and boms if they don't have the right dependencies in them.

File: spring-cloud-stream-docs/src/main/asciidoc/spring-cloud-stream-overview.adoc [hunk 1]
```diff
@@ -885,3 +885,97 @@ enabled or disabled using the `management.health.binders.enabled` property.
...
+[source]
+----
+cd GreetingSource
+java -jar target/GreetingSource-0.0.1-SNAPSHOT.jar --spring.cloud.stream.bindings.output.destination=mydest
+
+cd LoggingSink
+java -jar target/LoggingSink-0.0.1-SNAPSHOT.jar --server.port=8090 --spring.cloud.stream.bindings.input.destination=mydest
+----
+
+The different server port is avoid collisions of the http port used to service the boot actuator endpoints.
```
Comment (dsyer) [inline spring-cloud-stream-docs/src/main/asciidoc/spring-cloud-stream-overview.adoc]: This is a little confusing (for me anyway). The collision is between the two apps you are running, not between ports used by this app (which I might assume you mean).

Also note the missing "to".

**Comment by mbogoevici** [comment]
I believe that the version inconsistency is due to the Stream version of Brixton.M5 taking precedence over the Kafka binder starter.

File: spring-cloud-stream-docs/src/main/asciidoc/spring-cloud-stream-overview.adoc (hunk 1 above)
Comment (markpollack) [inline spring-cloud-stream-docs/src/main/asciidoc/spring-cloud-stream-overview.adoc]: Will fix, tx

File: spring-cloud-stream-docs/src/main/asciidoc/spring-cloud-stream-overview.adoc [hunk 2]
```diff
@@ -885,3 +885,97 @@ enabled or disabled using the `management.health.binders.enabled` property.
 == Samples
 
 For Spring Cloud Stream samples, please refer: https://github.com/spring-cloud/spring-cloud-stream-samples
+
+== Getting Started
+
+To get started creating Spring Cloud Stream applications, head over to https://start.spring.io and create a new project named `GreetingSource`.   Select the Spring Boot Version to be 1.3.1 and tick the checkbox for `Stream Kafka` as we will be using Kafka for messaging.  Next create a new class `GreetingSource` in the same package as the class `GreetingSourceApplication` with the following code
```
Comment (trisberg) [inline spring-cloud-stream-docs/src/main/asciidoc/spring-cloud-stream-overview.adoc]: Shouldn't the Boot version be 1.3.3?

Comment (mbogoevici) [inline spring-cloud-stream-docs/src/main/asciidoc/spring-cloud-stream-overview.adoc]: Actually, should be 1.3.4.BUILD-SNAPSHOT, to pick up up stuff after M4

**Comment by mbogoevici** [comment]
Merged w/https://github.com/spring-cloud/spring-cloud-stream/commit/7bf9215a6c38c8027635cea0068ea6c2a8b0d913

**Pull request closed without merging**